tests/system/test_multigroups/test_multigroups.py

```diff
@@ -0,0 +1,259 @@
+agent_groups = {'wazuh-agent1': ['default', 'test_mg_0'],
+                'wazuh-agent2': ['default', 'test_mg_1']}
+
+shared_folder_path = os.path.join(WAZUH_PATH, 'etc', 'shared')
+mg_folder_path = os.path.join(WAZUH_PATH, 'var', 'multigroups')
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'provisioning',
+                              'basic_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+token = host_manager.get_api_token(test_hosts[0])
+time_to_update = 10
+time_to_sync = 20
+
+
+# Functions
+
+def calculate_mg_name(groups_list):
+    """Determine multigroup name from a list of groups.
+
+    Args:
+        groups_list (list): List of groups that form a multigroup.
+
+    Returns:
+        str: Multigroup name.
+    """
+    return hashlib.sha256(','.join(groups_list).encode()).hexdigest()[:8]
+
+
+def get_mtime(folders_to_check, hosts=None):
+    """Get modification time of each file within the listed directories, for each listed host.
+
+    Args:
+        folders_to_check (iterable of str): Directories to recursively traverse to get the mtime of their files.
+        hosts (list): List of hosts whose folders should be iterated.
+
+    Returns:
+        result (dict): Dictionary with the host, path and modification time of each file.
+    """
+    hosts = test_hosts if hosts is None else hosts
+    result = {}
+
+    for host in hosts:
+        result[host] = {}
+        for folder in folders_to_check:
+            result[host].update(
+                {file['path']: file['mtime'] for file in host_manager.find_file(host, folder, recurse=True)['files']}
+            )
+
+    return result
+
+
+def get_agent_id(token, agent_name):
+    """Get agent ID from its name.
+
+    Args:
+        token (str): API token.
+        agent_name (str): Agent name.
+
+    Returns:
+        str: Agent ID.
+    """
+    agent_id = host_manager.make_api_call(host=test_hosts[0], token=token, method='GET',
+                                          endpoint=f"/agents?name={agent_name}")
+    assert agent_id['status'] == 200, f"Failed trying to get ID of agent {agent_name}"
+    return agent_id['json']['data']['affected_items'][0]['id']
+
+
+def delete_groups():
+    """Delete all groups, except Default"""
+    for agent_name, groups in agent_groups.items():
+        # Remove any pre-existing group.
+        host_manager.make_api_call(host=test_hosts[0], token=host_manager.get_api_token(test_hosts[0]), method='DELETE',
+                                   endpoint=f"/groups?groups_list={','.join(groups[1:])}")
+
+        # Remove any pre-existing multigroup (just in case they weren't removed after deleting the group).
+        mg_path = os.path.join(mg_folder_path, calculate_mg_name(groups))
+        for host in test_hosts:
+            if host_manager.run_shell(host, f"ls {mg_path}") == '':
+                break
+            host_manager.run_shell(host, cmd=f"rm -rf {mg_path}")
```

Comment:
Actually this is on purpose. If a file is not on the master, it will not exist on the workers either (they are synchronized). It is faster if the workers are not checked after that. 

If the file exists, it is deleted on all nodes to speed up the task, but eventually it would also be synchronized.